src/temperature/crud.py

```diff
@@ -0,0 +1,29 @@
+from sqlalchemy.orm import Session
+from .models import DBTemperature
+
+from .utils import get_temperature_for_all_cities, update_or_create_temperature
+
+
+def get_temperature_list(
+        db: Session,
+        skip: int,
+        limit: int,
+        city_id: int | None = None
+) -> list[DBTemperature]:
+    temperatures = db.query(DBTemperature)
+
+    if city_id:
+        temperatures = temperatures.filter(DBTemperature.city_id == city_id).first()
```

Comment:
> temperatures can be one or None, due to .first

yes, but only if there is the city_id argument, if not - it will return the list. So it has to be list[DBTemperature] | DBTempreture | None, right?

Reply:
if you want get list you should avoid `first()`

Reply:
delete this